### PR TITLE
Add mouse picking with bounds scaling + Update imguizmo

### LINKED ISSUE #211: Add mouse picking with the use of RmlUi onClick events
RmlUi supports throwing out events from UI code. We can leverage that to implement mouse picking.

### LINKED ISSUE #225: Add mouse picking by adding bounding boxes

### PR DESCRIPTION
**Pull request by IronicallySerious**
Fixes #225 
Closes #211

### CODE REVIEW

File: rootex/framework/components/transform_component.cpp [hunk 1]
```diff
@@ -7,10 +7,19 @@
 
 Component* TransformComponent::Create(const JSON::json& componentData)
 {
+	BoundingBox boundingBox({ 0.0f, 0.0f, 0.0f }, { 0.5f, 0.5f, 0.5f });
+	if (componentData.contains("boundingBox"))
```
Comment (sin3point14) [inline rootex/framework/components/transform_component.cpp:11]: i suppose this was for debugging?

File: rootex/framework/components/transform_component.h [hunk 2]
```diff
@@ -5,6 +5,14 @@
 
 class TransformComponent : public Component
 {
+public:
+	struct Bounds
```
Comment (sin3point14) [inline rootex/framework/components/transform_component.h:9]: I don't think this is being used anywhere

**Review by sin3point14 — CHANGES_REQUESTED** [review]
dorime plox 🙏

File: rootex/framework/components/visual/spot_light_component.cpp [hunk 3]
```diff
@@ -45,7 +45,7 @@ SpotLightComponent::SpotLightComponent(const float constAtt, const float linAtt,
     , m_DiffuseIntensity(diffuseIntensity)
     , m_Spot(spot)
 {
-	m_AngleRange = cos(angleRange * 3.1415 / 180);
+	m_AngleRange = angleRange;
```
Comment (sin3point14) [inline rootex/framework/components/visual/spot_light_component.cpp:48]: angle range has stopped working as shaders expect the cosine value to be there and this results in the number of degrees being sent to the shader

**Review by sin3point14 — COMMENTED** [review]
dorime plog again 🙏

File: rootex/framework/components/transform_component.cpp (hunk 1 above)
Comment (IronicallySerious) [reply rootex/framework/components/transform_component.cpp:11]: No, it's so that we don't have to edit component data manually

**Review by sin3point14 — APPROVED** [review]
dorime :pray:

**Pull request merged**